shared/middleware/tables_api.rb

```diff
@@ -280,6 +281,19 @@ def record_too_large(record_size, index = nil)
     return TableType.new(channel_id, storage_id(endpoint), table_name).to_csv
   end
 
+  # GET /v3/export-firebase-tables/<channel-id>/table-name
+  #
+  # Exports a csv file from a table where the first row is the column names
+  # and additional rows are the column values.
+  #
+  get %r{/v3/export-firebase-tables/([^/]+)/([^/]+)$} do |channel_id, table_name|
+    dont_cache
+    content_type :csv
+    response.headers['Content-Disposition'] = "attachment; filename=\"#{table_name}.csv\""
+
+    return FirebaseHelper.new(channel_id, table_name).table_as_csv
```

Comment:
Rails unencodes the table name for us. We have to encode it again before passing it in a url to the firebase wrapper in FirebaseHelper.